pravega/pravega
# Issue 8043: Client application stuck at initiateCheckpoint Api call

**Change log description**  
Client application get stuck when it triggers initiateCheckpoint API with Pravega internal executorService.
ReaderGroupManager constructor takes thread count as 1 while constructing SocketClientFactory, because of which initiatecheckpoint call get stuck.

**Purpose of the change**  
closes #7215 

**What the code does**  
Change thread count to 3 in SocketClientFactory while constructing ReaderGroupManager. 

**How to verify it**  
System test added for this change.
In system test added assertion for multiple checkpoint completion stage to make sure checkpoint is not blocked any more
